### PR TITLE
feat(React): adding a hot reloading React component

### PR DESCRIPTION
**Pull request by pmsorhaindo**
Adding a hot reloading React component transpiled by babel and bundled by webpack.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,13 @@
+node_modules
+npm-debug.log
+webpack-stats.json
+build/
+public/
+dist/
```
Comment (thatguynamedandy) [inline .gitignore]: can probably just limit it to `dist` at the moment, in the unlikely event that this catches up with us later on.

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+*.DS_Store
+*.log
+
+backend-status
+app-status
+client-status
```
Comment (thatguynamedandy) [inline .gitignore]: None of these are needed, in fact I sense this is just copied from elsewhere, maybe reset this to the minimal for now,

File: package.json [hunk 3]
```diff
@@ -4,12 +4,27 @@
...
     "test": ""
   },
   "repository": {
     "type": "git",
     "url": "git+https://github.com/BrandwatchLtd/brandwatch-react-app.git"
   },
   "license": "MIT",
-  "homepage": "https://github.com/BrandwatchLtd/brandwatch-react-app#readme"
+  "homepage": "https://github.com/BrandwatchLtd/brandwatch-react-app#readme",
+  "dependencies": {
```
Comment (thatguynamedandy) [inline package.json]: I know I helped decide on dev vs non dev dependencies. but with the static side rendering approach we maybe don't need to differentiate?

File: src/index.ejs [hunk 4]
```diff
@@ -0,0 +1,9 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta charset="UTF-8" />
+  </head>
```
Comment (HHogg) [inline src/index.ejs]: Would be nice to see the other standard head content here too...


```html
<meta content="width=device-width, initial-scale=1" name="viewport" />
```
> To handle optimising the view for mobile devices (sizing and zooming) 

```html
<title>Brandwatch | ReactApp</title>
```
> Basic title that can be changed once the app is forked. 

```html
<link href="https://fonts.googleapis.com/css?family=Roboto:100,300,400,500,700" rel="stylesheet">
```
> The standard Brandwatch application font set ... also required for Axiom

```html
<link href="/path/to/favicon.ico" rel="shortcut icon" />
```
> With the standard Brandwatch favicon

File: package.json (hunk 3 above)
Comment (HHogg) [inline package.json]: Also are we using `yarn` for this? (I can't see a `yarn.lock` file)

File: webpack.config.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    path: resolve(__dirname, 'dist'),
+  },
+  performance: {
+    hints: false,
+  },
+  plugins: [
+    new webpack.HotModuleReplacementPlugin(),
+    new webpack.NamedModulesPlugin(),
+    new HtmlWebpackPlugin({
+      template: 'index.ejs',
```
Comment (HHogg) [inline webpack.config.js]: You shouldn't need to use `.ejs` unless we're planning on dynamically adding stuff to the template. Which means you could also drop the `HtmlWebpackPlugin` and use `webpack`s content base directory to load an `index.html`.

File: webpack.config.js [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+    }),
+  ],
+  resolve: {
+    extensions: ['.js'],
```
Comment (HHogg) [inline webpack.config.js]: I don't this is needed, the default is `js` and `json`.

File: package.json [hunk 7]
```diff
@@ -4,12 +4,27 @@
...
+  },
+  "devDependencies": {
+    "babel": "^6.23.0",
+    "babel-core": "^6.26.0",
+    "babel-loader": "^7.1.2",
+    "babel-preset-es2015": "^6.24.1",
+    "babel-preset-react": "^6.24.1",
+    "html-webpack-plugin": "^2.30.1",
+    "webpack": "^3.5.5",
+    "webpack-dev-server": "^2.7.1"
```
Comment (HHogg) [inline package.json]: I think `react-hot-loader` is missing from this dependency list.

File: src/index.ejs (hunk 4 above)
Comment (thatguynamedandy) [inline src/index.ejs]: We were contemplating on using react Helmet for some of that. I think the font could be added when we add Axiom.

File: .gitignore (hunk 2 above)
Comment (pmsorhaindo) [inline .gitignore]: 👍

File: src/index.ejs (hunk 4 above)
Comment (pmsorhaindo) [inline src/index.ejs]: @thatguynamedandy and I were talking about looking at `react-helmet` to cover these things, future PR?

File: webpack.config.js [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+    './index.js',
+  ],
+  module: {
+    rules: [{
+      test: /\.js$/,
+      use: ['babel-loader'],
+      exclude: /node_modules/,
+    }, {
+      test: /\.ico$/,
+      loader: 'file-loader?name=[name].[ext]'
```
Comment (HHogg) [inline webpack.config.js]: I don't think this file-loader is needed for the ico... it looks like HtmlWebpackPlugin just inserts the tag with the given path.

**Pull request merged**